guides/manual_release.md

```diff
@@ -145,66 +145,6 @@ Example:
 
 # Creating a new Knative release using Prow
 
-**Note:** The user starting the release job must have privileged access to the
```

Comment:
Nice! I saw this one day and also think we should delete it.

Reply:
:)